src/test/java/StringManipulationTest.java

```diff
@@ -89,23 +122,102 @@ public void testGeSubStrings1() {
     @Test
     public void testGeSubStrings4() {
-        fail("Not yet implemented");
+    	//tests if IllegalArgumentException is thrown when startWord is zero
+    	manipulatedstring.setString("the quick brown fox jumps over the lazy dog");
+    	IllegalArgumentException thrown = assertThrows(IllegalArgumentException.class, () -> {
+    	String [] sStings = manipulatedstring.getSubStrings(0, 4);
+       	assertEquals(sStings[0], "the");
+       	assertEquals(sStings[1], "quick");
+       	assertEquals(sStings[2], "brown");
+       	assertEquals(sStings[4], "fox");
+       	assertEquals(sStings[5], "jumps");
+       	assertEquals(sStings[6], "over");
+       	assertEquals(sStings[7], "the");
+       	assertEquals(sStings[8], "lazy");
+       	assertEquals(sStings[3], "dog");
+  	  }, "Word position argument must be positive nonzero number.");
+    	
+    	assertEquals("Word position argument must be positive nonzero number.", thrown.getMessage());
     }
+    
     @Test
-    public void testGeSubStrings5() {
-        fail("Not yet implemented");
+    public void testGeSubStrings5() 
+    {
+    	manipulatedstring.setString("the quick brown fox jumps over the lazy dog");
+   	 	
+    	////tests if IndexOutOfBoundsException is thrown when arguments are greater than strings size
+    	IndexOutOfBoundsException thrown = assertThrows(IndexOutOfBoundsException.class, () -> {
+    	String [] sStings = manipulatedstring.getSubStrings(10, 11);
+    	assertEquals(sStings[0], "the");
+    	assertEquals(sStings[1], "quick");
+    	assertEquals(sStings[2], "brown");
+    	assertEquals(sStings[4], "fox");
+    	assertEquals(sStings[5], "jumps");
+    	assertEquals(sStings[6], "over");
+    	assertEquals(sStings[7], "the");
+    	assertEquals(sStings[8], "lazy");
+    	assertEquals(sStings[3], "dog");
+    	}, "upper bound or lower is greater than word count");
+        	
+        assertEquals("upper bound or lower is greater than word count", thrown.getMessage());
     }
+    
     @Test
     public void testGeSubStrings6() {
-        fail("Not yet implemented");
+    	//tests if IllegalArgumentException is thrown when endWord is negative
+    	manipulatedstring.setString("the quick brown fox jumps over the lazy dog");
+    	IllegalArgumentException thrown = assertThrows(IllegalArgumentException.class, () -> {
+    		String [] sStings = manipulatedstring.getSubStrings(-3, -1);
+   	 		assertEquals(sStings[0], "the");
+   	 		assertEquals(sStings[1], "quick");
+   	 		assertEquals(sStings[2], "brown");
+   	 		assertEquals(sStings[4], "fox");
+        	assertEquals(sStings[5], "jumps");
+        	assertEquals(sStings[6], "over");
+        	assertEquals(sStings[7], "the");
+        	assertEquals(sStings[8], "lazy");
+        	assertEquals(sStings[3], "dog");
+        	}, "Word position argument must be positive nonzero number.");
+            	
+            assertEquals("Word position argument must be positive nonzero number.", thrown.getMessage());
     }
 
```

Comment:
For getSubStrings() tests, because the behavior you're trying to verify is that the method throws an exception, the assertEquals statements may fall outside their scopes. The assertEquals() statements are best used in tests where you're testing the output, rather than the error checking.